src/BitNode/BitNodeUtils.ts

```diff
@@ -9,3 +14,56 @@ export function isBitNodeFinished(): boolean {
   }
   return wd.backdoorInstalled;
 }
+
+export function canAccessBitNodeFeature(bitNode: number): boolean {
+  return Player.bitNodeN === bitNode || Player.activeSourceFileLvl(bitNode) > 0;
+}
+
+export function knowAboutBitverse(): boolean {
+  return Player.bitNodeOptions.activeSourceFiles.size > 0;
+}
+
+export function getDefaultBitNodeOptions(): BitNodeOptions {
+  return {
+    activeSourceFiles: new Map(Player.sourceFiles),
+    restrictHomePCUpgrade: false,
+    disableGang: false,
+    disableCorporation: false,
+    disableBladeburner: false,
+    disable4SData: false,
+    disableHacknetServer: false,
+    disableSleeveExpAndAugmentation: false,
+  };
+}
+
+export function validateActiveSourceFiles(activeSourceFiles: Map<number, number>): {
+  valid: boolean;
+  message?: string;
+} {
+  for (const [sfNumber, sfLevel] of activeSourceFiles.entries()) {
+    if (!validBitNodes.includes(sfNumber)) {
+      return { valid: false, message: `Invalid BitNode: ${sfNumber}.` };
+    }
+    const maxSfLevel = Player.sourceFileLvl(sfNumber);
+    if (sfLevel > maxSfLevel) {
+      return { valid: false, message: `Invalid SF level: ${sfLevel}. Max level: ${maxSfLevel}.` };
+    }
+  }
+  return { valid: true };
+}
+
+export function setBitNodeOptions(bitNodeOptions: BitNodeOptions): void {
+  const validationResultForActiveSourceFiles = validateActiveSourceFiles(bitNodeOptions.activeSourceFiles);
+  if (!validationResultForActiveSourceFiles.valid) {
+    throw new Error(`activeSourceFiles is invalid. Reason: ${validationResultForActiveSourceFiles.message}`);
+  }
+
+  Player.bitNodeOptions.activeSourceFiles = new JSONMap(bitNodeOptions.activeSourceFiles);
```

Comment:
Why can't you do either `Player.bitNodeOptions = bitNodeOptions` or `Object.assign(Player.bitNodeOptions, bitNodeOptions)`?